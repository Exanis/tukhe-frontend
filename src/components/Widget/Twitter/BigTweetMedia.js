import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { withStyles } from '@material-ui/core/styles';

import Typography from '@material-ui/core/Typography';
import ArrowLeft from '@material-ui/icons/KeyboardArrowLeft';
import ArrowRight from '@material-ui/icons/KeyboardArrowRight';
import ConditionalComponent from "../../../components/ConditionalComponent/index";
import TweetUserName from './TweetUserName';
import TweetVideo from './TweetVideo';

export const classes = theme => ({
    bigMediaContainer: {
        height: '80vh',
    },
    bigMediaPicture: {
        width: '80%',
        flex: '0.8',
        floatLeft: 'auto',
        floatRight: 'auto',
        height: "100%",
        objectFit: 'contain'
    },
    bigMediaSubText: {
        width: '60%',
        marginLeft: '20%',
    },
    bigMediaPictureContainer: {
        width: "100%",
        height: 'calc(100% - 128px)',
        display: 'flex',
        alignItems: 'center'
    },
    bigMediaLink: {
        flex: '0.1',
        width: '10%',
        textAlign: 'center'
    },
    bigMediaShortcuts: {
        width: '60%',
        marginLeft: '20%',
        marginTop: 10,
        marginBottom: 10,
        display: 'flex',
        flexDirection: 'row'
    },
    bigMediaShortcutsLinkLeft: {
        fontSize: 12,
        color: theme.palette.primary.light,
        flex: 1
    },
    bigMediaShortcutsLinkRight: {
        fontSize: 12,
        color: theme.palette.primary.light,
    },
    tweetText: {
        wordBreak: 'break-word',
        '& a': {
            textDecoration: 'none',
            color: theme.palette.primary.main,
            '&:visited': {
                color: theme.palette.primary.light
            },
            '&:hover': {
                textDecoration: 'underline'
            }
        }
    },
});

export class RawBigTweetMedia extends Component {
    render() {
        return <div className={this.props.classes.bigMediaContainer}>
            <div className={this.props.classes.bigMediaPictureContainer}>
                <ConditionalComponent render={Boolean(this.props.mediaList[this.props.currentMedia - 1])}>
                    <a className={this.props.classes.bigMediaLink} onClick={this.props.displayMediaInPopup(
                        this.props.mediaList,
                        this.props.currentMedia - 1,
                        this.props.text,
                        this.props.author)}>
                        <ArrowLeft fontSize='large' />
                    </a>
                </ConditionalComponent>
                <ConditionalComponent render={!Boolean(this.props.mediaList[this.props.currentMedia - 1])}>
                    <div className={this.props.classes.bigMediaLink} />
                </ConditionalComponent>
                <ConditionalComponent render={this.props.media.type === 'photo'}>
                    <img src={`${this.props.media.media_url}:large`} className={this.props.classes.bigMediaPicture} />
                </ConditionalComponent>
                <ConditionalComponent render={this.props.media.type !== 'photo'}>
                    <TweetVideo gif={this.props.media.type === 'animated_gif'} video={this.props.media.video} className={this.props.classes.bigMediaPicture} />
                </ConditionalComponent>
                <ConditionalComponent render={Boolean(this.props.mediaList[this.props.currentMedia + 1])}>
                    <a className={this.props.classes.bigMediaLink} onClick={this.props.displayMediaInPopup(
                        this.props.mediaList,
                        this.props.currentMedia + 1,
                        this.props.text,
                        this.props.author)}>
                        <ArrowRight fontSize='large' />
                    </a>
                </ConditionalComponent>
                <ConditionalComponent render={!Boolean(this.props.mediaList[this.props.currentMedia + 1])}>
                    <div className={this.props.classes.bigMediaLink} />
                </ConditionalComponent>
            </div>
            <div className={this.props.classes.bigMediaShortcuts}>
                <a href={`${this.props.media.media_url}:large`} className={this.props.classes.bigMediaShortcutsLinkLeft}>
                    <FormattedMessage
                        id={"widget.twitter.tweet.viewOriginal"}
                        defaultMessage={'View image'}
                        />
                </a>
                <a href={`https://${this.props.media.display_url}`} className={this.props.classes.bigMediaShortcutsLinkRight}>
                    <FormattedMessage
                        id={"widget.twitter.tweet.viewImageTweet"}
                        defaultMessage={'View original tweet'}
                        />
                </a>
            </div>
            <div className={this.props.classes.bigMediaSubText}>
                <div className={this.props.classes.tweetText}>
                    <TweetUserName user={this.props.author}/>
                    <Typography variant='body1'>{this.props.text}</Typography>
                </div>
            </div>
        </div>
    }
}

const BigTweetMedia = withStyles(classes)(RawBigTweetMedia);

BigTweetMedia.propTypes = {
    mediaList: PropTypes.array,
    currentMedia: PropTypes.number.isRequired,
    displayMediaInPopup: PropTypes.func,
    text: PropTypes.string,
    author: PropTypes.object,
    media: PropTypes.shape({
        type: PropTypes.string,
        media_url: PropTypes.string,
        display_url: PropTypes.string
    })
};
RawBigTweetMedia.propTypes = {
    ...BigTweetMedia.propTypes,
    classes: PropTypes.object
};

export default BigTweetMedia;
