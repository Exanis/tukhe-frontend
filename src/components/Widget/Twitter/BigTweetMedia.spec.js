import React from 'react';
import { shallow } from 'enzyme';
import { classes, RawBigTweetMedia } from "./BigTweetMedia";

const props = {
    mediaList: [1, 2, 3],
    currentMedia: 0,
    displayMediaInPopup: jest.fn(),
    text: 'tweet text',
    author: {
        protected: true,
        verified: true,
        screen_name: 'test',
        name: 'test'
    },
    media: {
        type: 'media type',
        media_url: 'media url',
        display_url: 'display url',
        video: {
            variant: []
        }
    },
    classes: {
        bigMediaPicture: ''
    }
};

it('Should render correctly', () => {
    const wrapper = shallow(<RawBigTweetMedia {...props} />);

    expect(wrapper).toMatchSnapshot();
});

it('Should keep styles coherent', () => {
    const theme = {
        palette: {
            primary: {
                main: '#000',
                light: '#aaa'
            }
        }
    };

    expect(classes(theme)).toMatchSnapshot();
});
