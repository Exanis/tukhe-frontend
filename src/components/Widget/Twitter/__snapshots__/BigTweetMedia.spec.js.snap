// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should keep styles coherent 1`] = `
Object {
  "bigMediaContainer": Object {
    "height": "80vh",
  },
  "bigMediaLink": Object {
    "flex": "0.1",
    "textAlign": "center",
    "width": "10%",
  },
  "bigMediaPicture": Object {
    "flex": "0.8",
    "floatLeft": "auto",
    "floatRight": "auto",
    "height": "100%",
    "objectFit": "contain",
    "width": "80%",
  },
  "bigMediaPictureContainer": Object {
    "alignItems": "center",
    "display": "flex",
    "height": "calc(100% - 128px)",
    "width": "100%",
  },
  "bigMediaShortcuts": Object {
    "display": "flex",
    "flexDirection": "row",
    "marginBottom": 10,
    "marginLeft": "20%",
    "marginTop": 10,
    "width": "60%",
  },
  "bigMediaShortcutsLinkLeft": Object {
    "color": "#aaa",
    "flex": 1,
    "fontSize": 12,
  },
  "bigMediaShortcutsLinkRight": Object {
    "color": "#aaa",
    "fontSize": 12,
  },
  "bigMediaSubText": Object {
    "marginLeft": "20%",
    "width": "60%",
  },
  "tweetText": Object {
    "& a": Object {
      "&:hover": Object {
        "textDecoration": "underline",
      },
      "&:visited": Object {
        "color": "#aaa",
      },
      "color": "#000",
      "textDecoration": "none",
    },
    "wordBreak": "break-word",
  },
}
`;

exports[`Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RawBigTweetMedia
    author={
      Object {
        "name": "test",
        "protected": true,
        "screen_name": "test",
        "verified": true,
      }
    }
    classes={
      Object {
        "bigMediaPicture": "",
      }
    }
    currentMedia={0}
    displayMediaInPopup={
      [MockFunction] {
        "calls": Array [
          Array [
            Array [
              1,
              2,
              3,
            ],
            -1,
            "tweet text",
            Object {
              "name": "test",
              "protected": true,
              "screen_name": "test",
              "verified": true,
            },
          ],
          Array [
            Array [
              1,
              2,
              3,
            ],
            1,
            "tweet text",
            Object {
              "name": "test",
              "protected": true,
              "screen_name": "test",
              "verified": true,
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    media={
      Object {
        "display_url": "display url",
        "media_url": "media url",
        "type": "media type",
        "video": Object {
          "variant": Array [],
        },
      }
    }
    mediaList={
      Array [
        1,
        2,
        3,
      ]
    }
    text="tweet text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <ConditionalComponent
            render={false}
          >
            <a>
              <pure(KeyboardArrowLeftIcon)
                fontSize="large"
              />
            </a>
          </ConditionalComponent>
          <ConditionalComponent
            render={true}
          >
            <div />
          </ConditionalComponent>
          <ConditionalComponent
            render={false}
          >
            <img
              className=""
              src="media url:large"
            />
          </ConditionalComponent>
          <ConditionalComponent
            render={true}
          >
            <TweetVideo
              className=""
              gif={false}
              video={
                Object {
                  "variant": Array [],
                }
              }
            />
          </ConditionalComponent>
          <ConditionalComponent
            render={true}
          >
            <a>
              <pure(KeyboardArrowRightIcon)
                fontSize="large"
              />
            </a>
          </ConditionalComponent>
          <ConditionalComponent
            render={false}
          >
            <div />
          </ConditionalComponent>
        </div>,
        <div>
          <a
            href="media url:large"
          >
            <FormattedMessage
              defaultMessage="View image"
              id="widget.twitter.tweet.viewOriginal"
              values={Object {}}
            />
          </a>
          <a
            href="https://display url"
          >
            <FormattedMessage
              defaultMessage="View original tweet"
              id="widget.twitter.tweet.viewImageTweet"
              values={Object {}}
            />
          </a>
        </div>,
        <div>
          <div>
            <WithStyles(RawTweetUserName)
              user={
                Object {
                  "name": "test",
                  "protected": true,
                  "screen_name": "test",
                  "verified": true,
                }
              }
            />
            <WithStyles(Typography)
              variant="body1"
            >
              tweet text
            </WithStyles(Typography)>
          </div>
        </div>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ConditionalComponent
              render={false}
            >
              <a>
                <pure(KeyboardArrowLeftIcon)
                  fontSize="large"
                />
              </a>
            </ConditionalComponent>,
            <ConditionalComponent
              render={true}
            >
              <div />
            </ConditionalComponent>,
            <ConditionalComponent
              render={false}
            >
              <img
                className=""
                src="media url:large"
              />
            </ConditionalComponent>,
            <ConditionalComponent
              render={true}
            >
              <TweetVideo
                className=""
                gif={false}
                video={
                  Object {
                    "variant": Array [],
                  }
                }
              />
            </ConditionalComponent>,
            <ConditionalComponent
              render={true}
            >
              <a>
                <pure(KeyboardArrowRightIcon)
                  fontSize="large"
                />
              </a>
            </ConditionalComponent>,
            <ConditionalComponent
              render={false}
            >
              <div />
            </ConditionalComponent>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <a>
                <pure(KeyboardArrowLeftIcon)
                  fontSize="large"
                />
              </a>,
              "render": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <pure(KeyboardArrowLeftIcon)
                  fontSize="large"
                />,
                "className": undefined,
                "onClick": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "fontSize": "large",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "a",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <div />,
              "render": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <img
                className=""
                src="media url:large"
              />,
              "render": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "",
                "src": "media url:large",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <TweetVideo
                className=""
                gif={false}
                video={
                  Object {
                    "variant": Array [],
                  }
                }
              />,
              "render": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "gif": false,
                "video": Object {
                  "variant": Array [],
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <a>
                <pure(KeyboardArrowRightIcon)
                  fontSize="large"
                />
              </a>,
              "render": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <pure(KeyboardArrowRightIcon)
                  fontSize="large"
                />,
                "className": undefined,
                "onClick": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "fontSize": "large",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "a",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <div />,
              "render": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              href="media url:large"
            >
              <FormattedMessage
                defaultMessage="View image"
                id="widget.twitter.tweet.viewOriginal"
                values={Object {}}
              />
            </a>,
            <a
              href="https://display url"
            >
              <FormattedMessage
                defaultMessage="View original tweet"
                id="widget.twitter.tweet.viewImageTweet"
                values={Object {}}
              />
            </a>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FormattedMessage
                defaultMessage="View image"
                id="widget.twitter.tweet.viewOriginal"
                values={Object {}}
              />,
              "className": undefined,
              "href": "media url:large",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "defaultMessage": "View image",
                "id": "widget.twitter.tweet.viewOriginal",
                "values": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FormattedMessage
                defaultMessage="View original tweet"
                id="widget.twitter.tweet.viewImageTweet"
                values={Object {}}
              />,
              "className": undefined,
              "href": "https://display url",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "defaultMessage": "View original tweet",
                "id": "widget.twitter.tweet.viewImageTweet",
                "values": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "a",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            <WithStyles(RawTweetUserName)
              user={
                Object {
                  "name": "test",
                  "protected": true,
                  "screen_name": "test",
                  "verified": true,
                }
              }
            />
            <WithStyles(Typography)
              variant="body1"
            >
              tweet text
            </WithStyles(Typography)>
          </div>,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <WithStyles(RawTweetUserName)
                user={
                  Object {
                    "name": "test",
                    "protected": true,
                    "screen_name": "test",
                    "verified": true,
                  }
                }
              />,
              <WithStyles(Typography)
                variant="body1"
              >
                tweet text
              </WithStyles(Typography)>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "user": Object {
                  "name": "test",
                  "protected": true,
                  "screen_name": "test",
                  "verified": true,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "tweet text",
                "variant": "body1",
              },
              "ref": null,
              "rendered": "tweet text",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <ConditionalComponent
              render={false}
            >
              <a>
                <pure(KeyboardArrowLeftIcon)
                  fontSize="large"
                />
              </a>
            </ConditionalComponent>
            <ConditionalComponent
              render={true}
            >
              <div />
            </ConditionalComponent>
            <ConditionalComponent
              render={false}
            >
              <img
                className=""
                src="media url:large"
              />
            </ConditionalComponent>
            <ConditionalComponent
              render={true}
            >
              <TweetVideo
                className=""
                gif={false}
                video={
                  Object {
                    "variant": Array [],
                  }
                }
              />
            </ConditionalComponent>
            <ConditionalComponent
              render={true}
            >
              <a>
                <pure(KeyboardArrowRightIcon)
                  fontSize="large"
                />
              </a>
            </ConditionalComponent>
            <ConditionalComponent
              render={false}
            >
              <div />
            </ConditionalComponent>
          </div>,
          <div>
            <a
              href="media url:large"
            >
              <FormattedMessage
                defaultMessage="View image"
                id="widget.twitter.tweet.viewOriginal"
                values={Object {}}
              />
            </a>
            <a
              href="https://display url"
            >
              <FormattedMessage
                defaultMessage="View original tweet"
                id="widget.twitter.tweet.viewImageTweet"
                values={Object {}}
              />
            </a>
          </div>,
          <div>
            <div>
              <WithStyles(RawTweetUserName)
                user={
                  Object {
                    "name": "test",
                    "protected": true,
                    "screen_name": "test",
                    "verified": true,
                  }
                }
              />
              <WithStyles(Typography)
                variant="body1"
              >
                tweet text
              </WithStyles(Typography)>
            </div>
          </div>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ConditionalComponent
                render={false}
              >
                <a>
                  <pure(KeyboardArrowLeftIcon)
                    fontSize="large"
                  />
                </a>
              </ConditionalComponent>,
              <ConditionalComponent
                render={true}
              >
                <div />
              </ConditionalComponent>,
              <ConditionalComponent
                render={false}
              >
                <img
                  className=""
                  src="media url:large"
                />
              </ConditionalComponent>,
              <ConditionalComponent
                render={true}
              >
                <TweetVideo
                  className=""
                  gif={false}
                  video={
                    Object {
                      "variant": Array [],
                    }
                  }
                />
              </ConditionalComponent>,
              <ConditionalComponent
                render={true}
              >
                <a>
                  <pure(KeyboardArrowRightIcon)
                    fontSize="large"
                  />
                </a>
              </ConditionalComponent>,
              <ConditionalComponent
                render={false}
              >
                <div />
              </ConditionalComponent>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <a>
                  <pure(KeyboardArrowLeftIcon)
                    fontSize="large"
                  />
                </a>,
                "render": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <pure(KeyboardArrowLeftIcon)
                    fontSize="large"
                  />,
                  "className": undefined,
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "fontSize": "large",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "a",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <div />,
                "render": true,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <img
                  className=""
                  src="media url:large"
                />,
                "render": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "",
                  "src": "media url:large",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <TweetVideo
                  className=""
                  gif={false}
                  video={
                    Object {
                      "variant": Array [],
                    }
                  }
                />,
                "render": true,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "",
                  "gif": false,
                  "video": Object {
                    "variant": Array [],
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <a>
                  <pure(KeyboardArrowRightIcon)
                    fontSize="large"
                  />
                </a>,
                "render": true,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <pure(KeyboardArrowRightIcon)
                    fontSize="large"
                  />,
                  "className": undefined,
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "fontSize": "large",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "a",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <div />,
                "render": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                href="media url:large"
              >
                <FormattedMessage
                  defaultMessage="View image"
                  id="widget.twitter.tweet.viewOriginal"
                  values={Object {}}
                />
              </a>,
              <a
                href="https://display url"
              >
                <FormattedMessage
                  defaultMessage="View original tweet"
                  id="widget.twitter.tweet.viewImageTweet"
                  values={Object {}}
                />
              </a>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <FormattedMessage
                  defaultMessage="View image"
                  id="widget.twitter.tweet.viewOriginal"
                  values={Object {}}
                />,
                "className": undefined,
                "href": "media url:large",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "defaultMessage": "View image",
                  "id": "widget.twitter.tweet.viewOriginal",
                  "values": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <FormattedMessage
                  defaultMessage="View original tweet"
                  id="widget.twitter.tweet.viewImageTweet"
                  values={Object {}}
                />,
                "className": undefined,
                "href": "https://display url",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "defaultMessage": "View original tweet",
                  "id": "widget.twitter.tweet.viewImageTweet",
                  "values": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "a",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              <WithStyles(RawTweetUserName)
                user={
                  Object {
                    "name": "test",
                    "protected": true,
                    "screen_name": "test",
                    "verified": true,
                  }
                }
              />
              <WithStyles(Typography)
                variant="body1"
              >
                tweet text
              </WithStyles(Typography)>
            </div>,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <WithStyles(RawTweetUserName)
                  user={
                    Object {
                      "name": "test",
                      "protected": true,
                      "screen_name": "test",
                      "verified": true,
                    }
                  }
                />,
                <WithStyles(Typography)
                  variant="body1"
                >
                  tweet text
                </WithStyles(Typography)>,
              ],
              "className": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "user": Object {
                    "name": "test",
                    "protected": true,
                    "screen_name": "test",
                    "verified": true,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "tweet text",
                  "variant": "body1",
                },
                "ref": null,
                "rendered": "tweet text",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
